src/modules/microdds_client/dds_topics.h.em

```diff
@@ -0,0 +1,116 @@
+@###############################################
+@#
+@# EmPy template
+@#
+@###############################################
+@# Generates publications and subscriptions for XRCE
+@#
+@# Context:
+@#  - msgs (List) list of all RTPS messages
+@#  - topics (List) list of all topic names
+@#  - spec (msggen.MsgSpec) Parsed specification of the .msg file
+@###############################################
+@{
+import os
+
+
+}@
+
+#include <utilities.hpp>
+
+#include <uxr/client/client.h>
+#include <ucdr/microcdr.h>
+
+#include <uORB/Subscription.hpp>
+#include <uORB/Publication.hpp>
+@[for include in type_includes]@
+#include <uORB/ucdr/@(include).h>
+@[end for]@
+
+// Subscribers for messages to send
+struct SendTopicsSubs {
+@[    for pub in publications]@
+	uORB::Subscription @(pub['topic_simple'])_sub{ORB_ID(@(pub['topic_simple']))};
+	uxrObjectId @(pub['topic_simple'])_data_writer{};
+@[    end for]@
+
+	uint32_t num_payload_sent{};
+
+	void update(uxrSession *session, uxrStreamId create_stream_id, uxrStreamId publish_stream_id, uxrObjectId participant_id, const char *client_namespace);
+};
+
+void SendTopicsSubs::update(uxrSession *session, uxrStreamId create_stream_id, uxrStreamId publish_stream_id, uxrObjectId participant_id, const char *client_namespace)
+{
+	int64_t time_offset_us = session->time_offset / 1000; // ns -> us
```

Comment:
There is an issue here with synchronization. The time offset of the session is based on epoch while the timestamps are usually taken from monotonic time. So this shift doesn't work if you want to transform the timestamp between the companion computer and the PX4. A working solution is (maybe not the best):
`
int64_t time_offset_us = uxr_epoch_nanos(session) / 1000 - hrt_absolute_time();
`

Reply:
Did you see the commit this is carrying within XRCE-DDS to use hrt_absolute_time() for uxr_nanos()? 

https://github.com/PX4/Micro-XRCE-DDS-Client/commit/eff557dddbee5df5afe1da92d121d5f29d0cd006#

Reply:
I didn't see that but it is not the version of the submodule in this PR, is it?